src/Language/AST/Node.php

```diff
@@ -92,36 +93,29 @@ protected static function cloneValue($value)
 
     public function __toString(): string
     {
-        $tmp = $this->toArray(true);
-
-        return (string) json_encode($tmp);
+        return json_encode($this);
     }
 
     /**
      * @return array<string, mixed>
      */
-    public function toArray(bool $recursive = false): array
+    public function jsonSerialize(): array
     {
-        if ($recursive) {
-            return $this->recursiveToArray($this);
-        }
-
-        $tmp = (array) $this;
-
-        if ($this->loc !== null) {
-            $tmp['loc'] = [
-                'start' => $this->loc->start,
-                'end'   => $this->loc->end,
-            ];
-        }
+        return $this->toArray();
+    }
```

Comment:
Is this actually needed?

Reply:
Yeah, it makes the JSON representation more compact. In contrast to the default serialization, null values and some runtime only values (such as pointers to previous and next tokens in `Location`) are excluded.